readthedocs/search/models.py

```diff
@@ -0,0 +1,76 @@
+"""Search Queries."""
+
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+from django_extensions.db.models import TimeStampedModel
+
+from readthedocs.builds.models import Version
+from readthedocs.projects.models import Project
+from readthedocs.projects.querysets import RelatedProjectQuerySet
+
+
+class SearchQuery(TimeStampedModel):
+
+    """Information about the search queries."""
+
+    project = models.ForeignKey(
+        Project,
+        related_name='search_queries',
+        on_delete=models.CASCADE,
+    )
+    version = models.ForeignKey(
+        Version,
+        verbose_name=_('Version'),
+        related_name='search_queries',
+        on_delete=models.CASCADE,
+    )
+    query = models.CharField(
+        _('Query'),
+        max_length=4092,
+    )
+    count = models.PositiveIntegerField(
```

Comment:
I'm wondering if we want more data here. Should we be storing an object _each_ time a search happens? That way we can show the frequency of a search over time. Currently, this only tells us how many times a search has happened.

I think if we plan to delete the data every 3 months, we can probably store _every_ search query with it's own timestamp. I'm fine with shipping this initially though, before we start storing a lot more data.

Reply:
I am not sure what you meant be `more data`.

Storing search object everytime a search was made is a better idea. I realised that the graphs were wrong before. And going this way makes them correct and easier.

> That way we can show the frequency of a search over time.

Can you exapand this more?
* Do we want this to be selected by user? Like the user can select a date and we show him the frequency of searches made vs time for that day.
* Or we just show this for today/yesterday?

Reply:
Just that we will be able to see when a specific search was done each time it was done. The current modeling only shows the number of times a search was done, but no time data about each search.